src/Deserializers/FailingDeserializer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Wikibase\DataModel\Deserializers;
+
+use Deserializers\Deserializer;
+use Deserializers\Exceptions\DeserializationException;
+use Wikibase\DataModel\Term\Term;
+use Wikimedia\Assert\Assert;
+
+/**
+ * A Deserializer that fails when certain fields are present in the serialization.
+ * This can be used to prevent ingestion of certain data structures.
+ *
+ * @since 2.1
+ *
+ * @licence GNU GPL v2+
+ * @author Daniel Kinzler
+ */
+class FailingDeserializer implements Deserializer {
+
+	/**
+	 * @var string[]
+	 */
+	private $fields;
+
+	/**
+	 * @var string
+	 */
+	private $message;
+
+	/**
+	 * @param string[] $fields
```

Comment:
I find this class extremely confusing. It called a "failing" deserializer. I would expect it to **always** fail. But it does not when one of the fields is present (not absent but present) in the serialization. What's the use of this? The least thing you need to do is to rename the class (no idea to what) and the $fields array. But you can't name it $expectedFields, because it will not check all fields.

The whole thing feels like it should be a private method and not a class.